src/rgw/rgw_dedup_cluster.cc

```diff
@@ -719,41 +731,46 @@ namespace rgw::dedup {
         decode(sp, p);
       }
       catch (const buffer::error&) {
-        ldpp_dout(dpp, 0) << __func__ << "::failed shard_progress_t decode!" << dendl;
-        return false;
+        ldpp_dout(dpp, 1) << __func__ << "::failed shard_progress_t decode!" << dendl;
+        completed_arr[shard] = TOKEN_STATE_CORRUPTED;
+        continue;
```

Comment:
this is a behavior change  - not bailing out on a malformed token?

Reply:
The first pass collects state from all token, but will still fail the call.
After a wait time of 120 seconds the caller will pass "FORCE" option allowing it to skip passed failing tokens.
Active in-process tokens will still wait for completion.